atomicapp/providers/openshift.py

```diff
@@ -396,51 +399,59 @@ def _parse_kubeconf_data(self, kubecfg):
         token = user["user"]["token"]
         if "namespace" in context["context"]:
             namespace = context["context"]["namespace"]
-
-        return (url, token, namespace)
-
-    def _get_config_values(self):
+        if "insecure-skip-tls-verify" in cluster["cluster"]:
+            tls_verify = not cluster["cluster"]["insecure-skip-tls-verify"]
+        elif "certificate-authority" in cluster["cluster"]:
+            ca = cluster["cluster"]["certificate-authority"]
+
+        return {PROVIDER_API_KEY: url,
+                ACCESS_TOKEN_KEY: token,
+                NAMESPACE_KEY: namespace,
+                PROVIDER_TLS_VERIFY_KEY: tls_verify,
+                PROVIDER_CA_KEY: ca}
+
+    def _set_config_values(self):
         """
         Reads providerapi, namespace and accesstoken from answers.conf and
         corresponding values from providerconfig (if set).
         Use one that is set, if both are set and have conflicting values raise
         exception.
 
-        Returns:
-            tuple (providerapi, accesstoken, providerapi)
-
         Raises:
             ProviderFailedException: values in providerconfig and answers.conf
                 are in conflict
 
         """
-        result = {"namespace": self.namespace,
-                  "access_token": self.access_token,
-                  "providerapi": self.providerapi}
 
-        answers = {"namespace": None,
-                   "access_token": None,
-                   "providerapi": None}
-        providerconfig = {"namespace": None,
-                          "access_token": None,
-                          "providerapi": None}
+        # initialize result to default values
+        result = {PROVIDER_API_KEY: self.providerapi,
+                  ACCESS_TOKEN_KEY: self.access_token,
+                  NAMESPACE_KEY: self.namespace,
+                  PROVIDER_TLS_VERIFY_KEY: self.ssl_verify,
+                  PROVIDER_CA_KEY: None}
```

Comment:
I think it would still make sense to initialize `answers` and `providerconfig` up here:

```
for k in result.keys():
    answers[k] = None
    providerconfig[k] = None
```